wrs/ceph-init-wrapper.sh

```diff
@@ -214,30 +247,52 @@ status ()
                 done
                 echo "$text" | tr -d '\n' > $CEPH_STATUS_FAILURE_TEXT_FILE
             else
-               echo "$host: '${CEPH_SCRIPT} status' result contains invalid process names: $erred_procs"
-               echo "undetermined_osd" > $CEPH_STATUS_FAILURE_TEXT_FILE
+               echo "$host: '${CEPH_SCRIPT} status $1' result contains invalid process names: $erred_procs"
+               echo "Undetermined osd or monitor id" > $CEPH_STATUS_FAILURE_TEXT_FILE
             fi
         fi
+
         rm -f ${CEPH_GET_STATUS_FILE}
+
+        if [[ $RC == 0 ]] && [[ "$1" == "mon" ]] && [[ "$system_type" == "All-in-one" ]] && [[ "$system_mode" != "simplex" ]]
+        then
+            # SM needs exit code != 0 from 'status mon' argument of the init script on
+            # standby controller otherwise it thinks that the monitor is running and
+            # tries to stop it.
+            # '/etc/init.d/ceph status mon' checks the status of monitors configured in
+            # /etc/ceph/ceph.conf and if it should be running on current host.
+            # If it should not be running it just exits with code 0. This is what
+            # happens on the standby controller.
+            # When floating monitor is running on active controller /var/lib/ceph/mon of
+            # standby is not mounted (Ceph monitor partition is DRBD synced).
+            ls "/var/lib/ceph/mon/ceph-controller" > /dev/null 2>&1
```

Comment:
Assuming ceph-controller is a file, any reason that this isn't a std file test operation (if test  -e /var/lib/ceph/mon/ceph-controller) instead of ls?

Reply:
No reason,  did not think about "test -e". We'll replace it.